plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java

```diff
@@ -88,6 +90,23 @@ public Connection(IClient client, ICredentialsPrompter credentialsPrompter, ISSL
 		this.sslCertificateCallback = sslCertCallback;
 	}
 	
+	@Override
+	public Properties getExtendedProperties() {
+		return extendedProperties;
+	}
+
+	@Override
+	public void setExtendedProperty(String name, String value) {
+		Properties oldExt = new Properties(extendedProperties);
```

Comment:
Is this a concern for me?  I am just trying to make sure 'firePropertyChange' to actually propagates the event.  Should it really be send the original as old and then create a new one?